From feb3e1bebd7d3275b83e2a4ee87b7892da134887 Mon Sep 17 00:00:00 2001
Message-Id: <patch-1.9-feb3e1bebd7-20230202T093212Z-avarab@gmail.com>
In-Reply-To: <cover-0.9-00000000000-20230202T093212Z-avarab@gmail.com>
References: <230131.86357rrtsg.gmgdl@evledraar.gmail.com>
	<cover-0.9-00000000000-20230202T093212Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Wed, 1 Feb 2023 11:41:59 +0100
Subject: [PATCH 1/9] archive & tar config docs: de-duplicate configuration
 section
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The "tar.umask" documentation was initially added in [1], and was
duplicated from the start. Then with [2] the two started drifting
apart. Let's consolidate them with a change like the ones made in the
commits merged in [3].

1. ce1a79b6a74 (tar-tree: add the "tar.umask" config option,
   2006-07-20)
2. 687157c736d (Documentation: update tar.umask default, 2007-08-21)
3. 7a54d740451 (Merge branch 'ab/dedup-config-and-command-docs',
   2022-09-14)

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 Documentation/config/tar.txt  | 4 +++-
 Documentation/git-archive.txt | 8 +-------
 2 files changed, 4 insertions(+), 8 deletions(-)

diff --git a/Documentation/config/tar.txt b/Documentation/config/tar.txt
index de8ff48ea9d..c68e294bbc5 100644
--- a/Documentation/config/tar.txt
+++ b/Documentation/config/tar.txt
@@ -3,4 +3,6 @@ tar.umask::
 	tar archive entries.  The default is 0002, which turns off the
 	world write bit.  The special value "user" indicates that the
 	archiving user's umask will be used instead.  See umask(2) and
-	linkgit:git-archive[1].
+	linkgit:git-archive[1] for
+	details. If `--remote` is used then only the configuration of
+	the remote repository takes effect.
diff --git a/Documentation/git-archive.txt b/Documentation/git-archive.txt
index 60c040988bb..bbb407d4975 100644
--- a/Documentation/git-archive.txt
+++ b/Documentation/git-archive.txt
@@ -131,13 +131,7 @@ tar
 CONFIGURATION
 -------------
 
-tar.umask::
-	This variable can be used to restrict the permission bits of
-	tar archive entries.  The default is 0002, which turns off the
-	world write bit.  The special value "user" indicates that the
-	archiving user's umask will be used instead.  See umask(2) for
-	details.  If `--remote` is used then only the configuration of
-	the remote repository takes effect.
+include::config/tar.txt[]
 
 tar.<format>.command::
 	This variable specifies a shell command through which the tar
-- 
2.39.1.1392.g63e6d408230

